fix(sysdig-deploy): Chart documentation fix values missing from configuration but referenced in examples (#811)
sysdiglabs/charts: charts/sysdig-deploy/README.md

## Configuration

> The following table lists the configurable parameters of this chart and their default values, it is indicate and is not complete. If you are looking for more advanced configuration values, those are available in the documentation of the admission-controller, agent, node-analyzer, kspm-collector and rapid-response charts.

Reviewer: ```suggestion
The following table lists the configurable parameters of this chart and their default values, it is indicative and contains only the most commonly used parameters. If you are looking for more advanced configuration values, those are available in the documentation of the admission-controller, agent, node-analyzer, kspm-collector and rapid-response charts.
```

> The following table lists the configurable parameters of this chart and their default values, it is indicative and contains only the most commonly used parameters. If you are looking for more advanced configuration values, those are available in the documentation of the admission-controller, agent, node-analyzer, kspm-collector and rapid-response charts.

Reviewer 2: If I recall correctly, the only parameters/values we should be putting in the `sysdig-deploy` chart are those that can be used by multiple charts, or a simple boolean to enable/disable a given subchart. As such, I would recommend rephrasing this whole block similar to the following:

The following table lists the configurable parameters of the sysdig-deploy chart and their default values.  The sysdig-deploy chart itself only has select parameters that are used by multiple subcharts, and those used to enable/disable select subcharts. If you need additional configuration values, those are available in the various READMEs of the individual subcharts.

If you wanted to get fancy you could add links to the existing READMEs too.

Author: Thanks, I included your suggestion and added links as well :)

| `nodeAnalyzer`                          | Config specific to the [Sysdig nodeAnalyzer](#nodeanalyzer)                                                             | `{}`      |
| `nodeAnalyzer.enabled`                  | Enable the nodeAnalyzer component in this chart                                                                         | `true`    |
| `nodeAnalyzer.secure.enabled`           | Enable Sysdig Secure                                                                                                    | `true`    |
| `nodeAnalyzer.secure.vulnerabilityManagement.newEngineOnly` | Enable only the new vulnerability management engine                                                 | `false`   |
| `nodeAnalyzer.nodeAnalyzer.apiEndpoint` | nodeAnalyzer apiEndpoint                                                                                                | `""`      |
| `nodeAnalyzer.nodeAnalyzer.benchmarkRunner.deploy` | Deploy the Benchmark Runner Scanner                                                                          | `true`   |

> | `nodeAnalyzer.nodeAnalyzer.runtimeScanner.deploy` | Deploy the Runtime Scanner                                                                                    | `false`   |

Reviewer 3: I remember Christian gave a presentation where some of these components require other values to be enabled or dismissed in order to work. @claffin

Reviewer 2: Based on the precedent of the chart at it stands, I don't believe these parameters should be documented as a part of the `sysdig-deploy` chart. It is my understanding that the sysdig-deploy chart is only supposed to contain values that can be used by multiple subcharts ,and the simple boolean flags to enable/disable various subcharts. Any subchart specific configuration needs to be contained, and thus documented, within the subchart itself to conform with Helm parent/subchart best practices: https://helm.sh/docs/chart_template_guide/subcharts_and_globals/.

Author: Hi @aroberts87 I understand the intent but the current state of the charts implementation doesn't allow it imho. The example commands we have in the readme do specifically reference those values, in which case it doesn't make much sense to not have them documented as well.

From our experience, we had to jump between the configuration of the different sub-charts and sometimes go inside the values definition and templates to understand how we could fix issues we had. I think Sysdig's users would benefit from more details, especially as more and more use only the sysdig-deploy chart and won't deploy other components individually.

The helm page about subcharts and globals, doesn't necessarily says that you shouldn't use subchart config:
> A parent chart can override values for subcharts.

For what it's worth, there is an automated way to generate the readme from the values file and then make sure values for subcharts are synced with the parent chart. 

Here is an example that bitnami is using: https://github.com/bitnami-labs/readme-generator-for-helm

Reviewer 2: I certainly understand your points @yilmi, and agree that managing these values and documentation about them while spanning multiple teams is troublesome. IMHO the fact that we have to specify subchart overrides in the base `sysdig-deploy` installation instructions itself is an issue though and we should first look there. That said, the reason I'm being a bit of a stick in the mud about this is because of the ongoing effort by @claffin to reduce confusion in the helm docs. I'll defer to his call here.

Author: @aroberts87 thanks for taking the time to provide some more background as I wasn't aware of this work. I get that you may want to offer a better experience with global or high-level values for the main chart. I'd be happy to share more details about the challenges we faced, see if that could help you evaluate the relevance of some values.
